services/FeefoReviewService.php

```diff
@@ -29,19 +27,22 @@ public function feefoMerchantId() {
 
   private function getProducts() {
     $cacheKey = 'feefo_'.$this->feefoMerchantId();
-    $products = craft()->cache->get($cacheKey);
-    if ($products === false) {
-      $products = $this->getProductReviews();
-      if (!is_null($products))
-        craft()->cache->set($cacheKey, $products, 24*3600 /*24 hours*/);
+    $products = $this->getProductReviews();
+    if (!is_null($products)) {
+      craft()->cache->set($cacheKey, $products, 24*3600 /*24 hours*/);
+      return $products;
+    } else {
+      return null;
     }
-    return $products;
   }
 
-  private function getProductReviews()
-  {
-    $data = file_get_contents('https://api.feefo.com/api/10/products/ratings?merchant_identifier='.$this->feefoMerchantId().'&review_count=true&since_period=year');
-    $decodedData = json_decode($data);
-    return $decodedData->products;
+  private function getProductReviews() {
+    $data = @file_get_contents('https://api.feefo.com/api/10/products/ratings?merchant_identifier='.$this->feefoMerchantId().'&review_count=true&since_period=year');
+    $decodedData = @json_decode($data);
+    if(isset($decodedData) && isset($decodedData->products)){
+      return $decodedData->products;
+    } else {
+      return null;
```

Comment:
If you return an empty array instead of a null you do not need check if products is null on line 16 or line 31. This removes the need for changing `getReviewSummary()` at all.